pipeline/stage05_wave_characterization/scripts/local_macro_observables.py

```diff
@@ -0,0 +1,103 @@
+import neo
+import numpy as np
+import quantities as pq
+import matplotlib.pyplot as plt
+import os
+import argparse
+import scipy
+import pandas as pd
+from utils import load_neo, none_or_str, save_plot, AnalogSignal2ImageSequence
+import math 
+
+
+def calc_point_velocities(evts, DIM_X, DIM_Y):
+    N_Waves = len(np.unique(evts.labels)) 
+    WaveColl = np.ones([N_Waves, DIM_X, DIM_Y])*np.nan
+    WaveColl[np.int32(np.float64(evts.labels)), np.int32(evts.array_annotations['x_coords']), np.int32(evts.array_annotations['y_coords'])] = evts.times
+    WaveColl_2d = np.reshape(WaveColl, [np.shape(WaveColl)[0], np.shape(WaveColl)[1]*np.shape(WaveColl)[2]])
+   
+    #compute velocity
+    Tx = np.diff(WaveColl, axis = 1)[:, 0:np.shape(WaveColl)[1]-1, 0:np.shape(WaveColl)[2]-1]
+    Ty = np.diff(WaveColl, axis = 2)[:, 0:np.shape(WaveColl)[1]-1, 0:np.shape(WaveColl)[2]-1]
+
+    Tx = np.reshape(Tx, [np.shape(Tx)[0], (np.shape(Tx)[1])*np.shape(Tx)[2]])
+    Ty = np.reshape(Ty, [np.shape(Ty)[0], np.shape(Ty)[1]*(np.shape(Ty)[2])])
+ 
+    vel =  np.sqrt((Tx**2 + Ty**2)/(2*evts.annotations['spatial_scale'])**2)*evts.times.units
+    vel = 1./vel
```

Comment:
You are calculating the velocity as sqrt((dx^2 + dy^2)/ (dt_x^2 + dt_y^2)), with dx=dy; while I believe it must be sqrt( (dx/dt_x)**2 + (dy/dt_y)^2 )